test_python/test_c2.py

```diff
@@ -35,10 +35,17 @@ def matmul(float(M,N) A, float(N,K) B) -> (output) {
 }
 """
 
+MATMUL_GRAD_LANG = """
```

Comment:
The cpp portions of the commit make sense.

For the python part, depending on whether or not the test can run in OSS, it should either be completely omitted if it doesn't run (i.e. it is dead code) or it should be also added to the CI in the same PR.

Commit message is off.

In any case a C++ test should be added to test_caffe2.cc which tests gradient from C++, otherwhise this is dead code.